Doc/reference/expressions.rst

```diff
@@ -1766,6 +1766,12 @@ Or, when processing a file stream in chunks:
    while chunk := file.read(9000):
        process(chunk)
 
+Assignment expressions must be surrounded by parentheses when used
+as sub-expressions in slicing, conditional, lambda,
```

Comment:
Unparenthesized assignment expressions in slices were implemented in #23317, so they are allowed since 3.10. Also, assignment expression need to parenthesized in `assert` and `with` statements.

Reply:
Unparenthesized walruses are allowed in index expressions but not slices. This is current 3.12:
```
>>> x[y:=3]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'x' is not defined
>>> x[y:=3:z:=4]
  File "<stdin>", line 1
    x[y:=3:z:=4]
          ^
SyntaxError: invalid syntax
```

Reply:
Correct! I mixed up indexes/slices. Sorry!